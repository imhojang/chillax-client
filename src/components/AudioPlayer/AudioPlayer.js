import React, { useRef, useEffect } from 'react';

const AudioPlayer = props => {
  const audioElement = useRef(null);

  useEffect(() => {
    if (props.isPlayingSound) {
      audioElement.current.play();
    } else {
      audioElement.current.pause();
    }
  }, [props.isPlayingSound]);

  useEffect(() => {
    audioElement.current.volume = props.volume;
  }, [props.volume]);

  return (
    <div>
      <audio
        id='default-audioplayer'
        src='https://rainymood.com/audio1110/0.m4a'
        preload='auto'
        loop
        ref={audioElement}
      />
    </div>
  );
};

export default AudioPlayer;
