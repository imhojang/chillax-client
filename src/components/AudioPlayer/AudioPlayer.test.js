import React from 'react';
import { shallow } from 'enzyme';
import AudioPlayer from './AudioPlayer';

describe('<AudioPlayer />', () => {
  let component = null;
  it('renders correctly', () => {
    component = shallow(<AudioPlayer />)
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  })

});