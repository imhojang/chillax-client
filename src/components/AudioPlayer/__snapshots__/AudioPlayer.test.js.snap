// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioPlayer /> matches snapshot 1`] = `
<div
  className="audio-player"
>
  <audio
    id="default-audioplayer"
    loop={true}
    preload="auto"
    src=""
  />
</div>
`;
